### PR TITLE
feat(delete-snapshots): Delete snapshots with data drives

### PR DESCRIPTION
**Pull request by ksendart**
#871

### CODE REVIEW

File: src/app/reducers/volumes/redux/volumes.effects.ts [hunk 1]
```diff
@@ -220,17 +222,26 @@ export class VolumesEffects {
...
+      this.dialog.open(VolumeDeleteDialogComponent, {
+        data: !!volumes.find(volume => !!volume.snapshots.length)
+      }).afterClosed()
         .filter(res => Boolean(res))
-        .flatMap(() => volumes
-          .map((volume: Volume) => new volumeActions.DeleteVolume(volume))));
+        .flatMap((params) => {
+          return volumes
+            .map((volume: Volume) => {
+              if (params.deleteSnapshots && !!volume.snapshots.length) {
```
Comment (zolotyx) [inline src/app/reducers/volumes/redux/volumes.effects.ts]: do not dispatch actions in effects

File: src/app/shared/actions/volume-actions/volume-actions.container.ts [hunk 2]
```diff
@@ -39,6 +39,12 @@ export class VolumeActionsContainerComponent {
       .onErrorResumeNext()
       .filter(res => Boolean(res))
       .subscribe(() => {
+        if (!!volume.snapshots.length) {
```
Comment (zolotyx) [inline src/app/shared/actions/volume-actions/volume-actions.container.ts]: Do we really need to show 2 different dialogs here?

File: src/app/shared/actions/volume-actions/volume-delete/volume-delete-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  constructor(
+    public dialogRef: MatDialogRef<VolumeDeleteDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public hasSnapshots: boolean
+  ) { }
+
+  public confirmDestroy(): void {
+    const result = {};
+    if (this.deleteSnapshots) {
+      result['deleteSnapshots'] = true;
```
Comment (zolotyx) [inline src/app/shared/actions/volume-actions/volume-delete/volume-delete-dialog.component.ts]: remove square brackets, use typing {deleteSnapshots?: boolean}

File: src/app/shared/actions/volume-actions/volume-actions.container.ts (hunk 2 above)
Comment (ksendart) [reply src/app/shared/actions/volume-actions/volume-actions.container.ts]: We have the mock. I think it should be discussed with PM and PO

Comment (ershovaelena) [reply src/app/shared/actions/volume-actions/volume-actions.container.ts]: Yes, it was discussed with PO. We need 2 dialogs here (just the same that we have for deleted VMs)

**Pull request merged**